lib/matplotlib/text.py

```diff
@@ -2398,6 +2415,26 @@ def get_window_extent(self, renderer=None):
 
         return Bbox.union(bboxes)
 
+    def _set_to_best_position(self, renderer):
+        if self.axes is None:
+            return
+        bbox, info, descent = self._get_layout(renderer)
+        parent_bbox = self._get_bbox_to_anchor()
+        fontsize = renderer.points_to_pixels(self.get_size())
+        container = parent_bbox.padded(-0.5 * fontsize)
```

Comment:
Can `_get_bbox_to_anchor()` return `None` even if `self.axes is not None` (which looks like is guaranteed from your `if` check above on L2419)? if so then do we need another `None` check, or would that just be a horribly wrong case where throwing `AttributeError` is okay?

it's probably all fine, just wanted to point out the duplicate `None` check for `self.axes` you have. I guess that's a good idea since anyone else can start calling the `_get_bbox_to_anchor` helper

Reply:
Yeah I think it's worth having as a safety, but I agree it's currently not useful.